company_website/static/main_page/scripts/project-highlight.js

```diff
@@ -0,0 +1,132 @@
+    // ".portfolio-container",
+    // ".clutch-container",
+    // ".ready-to-work-with-us",
+    // ".map",
+    // ".footer-section",
+]
+
+projectBoxes.each(function() {
+    $(this).click(function() {
+        if(isWindowLarge())
+            disableScroll()
+        $(projectHighlightOverlay).toggle();
+        $(projectHighlightOverlay).toggleClass("non-clickable");
+        projectBoxes.toggleClass("non-clickable");
+        backgroundTransition();
+
+        let absoluteGridPosition = this.getBoundingClientRect();
+        projectHighlightOverlay.appendChild(this);
+        changeElementSize(this, absoluteGridPosition.width, absoluteGridPosition.height);
+        let absoluteHighlightPosition = this.getBoundingClientRect();
+        animateProjectHighlight(this, absoluteGridPosition, absoluteHighlightPosition);
+    });
+});
+
+$(projectHighlightOverlay).click(function() {
+    let originalContainer = findEmptyProjectHolder();
+    let absoluteGridPosition = originalContainer.getBoundingClientRect();
+    let highlightedProject = projectHighlightOverlay.getElementsByClassName("project-box")[0];
+
+    $(projectHighlightOverlay).toggleClass("non-clickable");
+    matchSizeToOtherProjectBoxes(highlightedProject);
+
+    let transitionEndListener = function() {
+        let absoluteHighlightPosition = highlightedProject.getBoundingClientRect();
+        backgroundTransition()
+        animateProjectReturn(highlightedProject, absoluteGridPosition, absoluteHighlightPosition, originalContainer)
+        highlightedProject.removeEventListener('transitionend', transitionEndListener);
+    }
+
+    highlightedProject.addEventListener('transitionend', transitionEndListener);
+    $(highlightedProject).toggleClass("expanded-project-box");
+});
+
+function animateProjectHighlight(projectElement, absoluteGridPosition, absoluteHighlightPosition) {
+    let fromParams = {
+        x: (isWindowLarge() ? absoluteGridPosition.left - absoluteHighlightPosition.left : 0),
+        y: absoluteGridPosition.top - absoluteHighlightPosition.top,
+    };
+    let toParams = {
+        x: 0,
+        y: 0,
+        duration: 0.5,
+        ease: "power3.out",
+        onCompleteParams: [projectElement],
+        onComplete: projectBoxExpansion,
+    };
+    gsap.fromTo(projectElement, fromParams, toParams);
+}
+
+function animateProjectReturn(projectElement, absoluteGridPosition, absoluteHighlightPosition, originalContainer) {
+    let fromParams = {
+        x: 0,
+        y: 0,
+    };
+    let toParams = {
+        x: (isWindowLarge() ? absoluteGridPosition.left - absoluteHighlightPosition.left : 0),
+        y: absoluteGridPosition.top - absoluteHighlightPosition.top,
+        duration: 0.5,
+        ease: "expo.out",
+        onCompleteParams: [projectElement, originalContainer],
+        onComplete: projectBoxReturn,
+    };
+    gsap.fromTo(projectElement, fromParams, toParams);
+}
+
+function projectBoxExpansion(projectElement) {
```

Comment:
This method name sounds for me like infinitive phrase. I think it should has verb inside. For example "expandProjectBox" or something like this

Reply:
Done.